File: app/src/main/java/com/mattieapps/clearnote/db/NoteDatabase.java
package com.mattieapps.clearnote.db;

import android.arch.persistence.room.Database;
import android.arch.persistence.room.Room;
import android.arch.persistence.room.RoomDatabase;
import android.content.Context;

import com.mattieapps.clearnote.model.Note;

/**
 * Created by andrewmattie on 12/17/17.
 */

@Database(entities = Note.class, version = 1)
public abstract class NoteDatabase extends RoomDatabase {
    private static NoteDatabase INSTANCE;

    public abstract NoteDao noteDao();

    public static NoteDatabase getNoteDatabase(Context context) {
        if (INSTANCE == null) {
            //TODO remove allowMainThreadQueries after testing is complete
            INSTANCE = Room.databaseBuilder(context.getApplicationContext(), NoteDatabase.class, "note-database").allowMainThreadQueries().build();
        }

        return INSTANCE;
    }

    public static void destroyInstance() {
        INSTANCE = null;
    }
}

File: app/src/main/java/com/mattieapps/clearnote/db/NoteDatabase.kt
package com.mattieapps.clearnote.db

import android.arch.persistence.room.Database
import android.arch.persistence.room.Room
import android.arch.persistence.room.RoomDatabase
import android.content.Context

import com.mattieapps.clearnote.model.Note

/**
 * Created by andrewmattie on 12/17/17.
 */

@Database(entities = [(Note::class)], version = 1)
abstract class NoteDatabase : RoomDatabase() {

    abstract fun noteDao(): NoteDao

    companion object {
        private var INSTANCE: NoteDatabase? = null

        fun getNoteDatabase(context: Context): NoteDatabase {
            if (INSTANCE == null) {
                INSTANCE = Room.databaseBuilder(context.applicationContext, NoteDatabase::class.java, "note-database").build()
            }

            return INSTANCE!!
        }

        fun destroyInstance() {
            INSTANCE = null
        }
    }
}
